google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/SingleUseTransaction.java

```diff
@@ -417,6 +419,30 @@ private ApiFuture<ResultSetStats> executeTransactionalUpdateAsync(
         ImmutableList.of(SpannerGrpc.getExecuteSqlMethod(), SpannerGrpc.getCommitMethod()));
   }
 
+  private ApiFuture<ResultSet> executeDmlReturningAsync(
+      final ParsedStatement update, QueryOption... options) {
+    Callable<ResultSet> callable =
+        () -> {
+          try {
+            writeTransaction = createWriteTransaction();
+            ResultSet resultSet =
+                writeTransaction.run(
+                    transaction ->
+                        DirectExecuteResultSet.ofResultSet(
+                            transaction.executeQuery(update.getStatement(), options)));
+            state = UnitOfWorkState.COMMITTED;
+            return resultSet;
+          } catch (Throwable t) {
+            state = UnitOfWorkState.COMMIT_FAILED;
+            throw t;
+          }
+        };
+    return executeStatementAsync(
+        update,
+        callable,
+        ImmutableList.of(SpannerGrpc.getExecuteSqlMethod(), SpannerGrpc.getCommitMethod()));
```

Comment:
Hmm.... This is going to be a little bit of a difficult decision:
1. `executeUpdate` used to use the `executeSql` method, while `executeQuery` used the `executeStreamingSql`. Now all statements will use `executeStreamingSql`.
2. Users can set custom timeout and retry settings per RPC. This meant that they previously could in theory set a different timeout value for queries and DML statements. That is no longer possible, as they use the same RPC.
3. So users who currently set a timeout value for `ExecuteSql` will now have to set a timeout value for `ExecuteStreamingSql`.

It also means that we need to replace all references to `SpannerGrpc.getExecuteSqlMethod()` with `SpannerGrpc.getExecuteStreamingSqlMethod()`.

Reply:
Done.